frontend/src/hooks/useIsFirstRender.ts

```diff
@@ -0,0 +1,11 @@
+import { useRef, useEffect } from 'react';
+
+function UseIsFirstRender() {
+	const isMountRef = useRef(true);
+	useEffect(() => {
+		isMountRef.current = false;
+	}, []);
+	return isMountRef.current;
+}
```

Comment:
Feels a bit overkill to extract this into its own hook. All you're really doing is storing a ref to indicate when the component has mounted.

Reply:
I see no reason to not extract self-contained logic, even if it is trivial. Especially if un-extracting it will take extra effort. Do you think it is less readable this way?

Reply:
There are plenty of reasons not to extract self-contained logic, particularly into its own file. This one is used in just one component, which is my primary argument. It is no _more_ readable sitting in its own file, than existing as a useRef in the same component that needs it.

Fragmenting your code is something to do when it aids in understanding and readability, because your code is becoming complex. But even in that case, there is rarely much benefit in extracting small pieces of code; instead you will benefit most from separating concerns.

I would like you to learn that refactoring can make things worse; keep related code co-located until such time as you find yourself repeating / copying logic. I am a strong advocate for keeping things simple; the Three Strikes rule is a good one to follow in cases like this:
https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)

Reply:
Interesting... This feels like something I'd have to learn the hard way to properly internalise it, but I'll take it on face value for now!

Reply:
I _have_ learned the hard way ;)

Reply:
_(Chris keeps quietly spreading the word about ReScript)_